File: codes/python/p009.py
# 
# Solution to Project Euler problem 9
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 


# Computers are fast, so we can implement a brute-force search to directly solve the problem.
def compute():
	PERIMETER = 1000
	for a in range(1, PERIMETER + 1):
		for b in range(a + 1, PERIMETER + 1):
			c = PERIMETER - a - b
			if a * a + b * b == c * c:
				# It is now implied that b < c, because we have a > 0
				return str(a * b * c)


if __name__ == "__main__":
	print(compute())

File: codes/Haskell/p009.hs
{- 
 - Solution to Project Euler problem 9
 - Copyright (c) Sankha Saha. All rights reserved.
 - 
 -  
 -   
 -}


-- Computers are fast, so we can implement a brute-force search to directly solve the problem.
perim = 1000
main = putStrLn (show ans)
ans = head [a * b * (perim - a - b) | a <- [1..perim], b <- [a+1..perim], isIntegerRightTriangle a b]
isIntegerRightTriangle a b = a < b && b < c
	&& a * a + b * b == c * c
	where c = perim - a - b
